crates/cairo-profiler/src/sierra_loader.rs

```diff
@@ -125,6 +145,30 @@ pub fn compile_sierra_and_add_compiled_artifacts_to_cache(
     Ok(())
 }
 
+fn maybe_get_statements_functions_map(
+    maybe_sierra_program_debug_info: Option<DebugInfo>,
+) -> Option<StatementsFunctionsMap> {
+    maybe_sierra_program_debug_info
+        .and_then(|mut debug_info| {
+            debug_info
+                .annotations
+                .shift_remove("github.com/software-mansion/cairo-profiler")
+        })
+        .map(|mut annotations| {
+            assert!(
+                annotations.get("statements_functions").is_some(),
+                "Wrong debug info annotations format"
+            );
+            annotations["statements_functions"].take()
+        })
+        .map(|statements_functions| {
+            let map =
+                serde_json::from_value::<HashMap<StatementIdx, Vec<String>>>(statements_functions)
+                    .expect("Wrong statements function map format");
+            StatementsFunctionsMap(map)
+        })
+}
```

Comment:
I don't know when its Some and when it's none. Please create a sub function which returns the data always and in `maybe_get_statements_functions_map` store only the `maybe` logic

Reply:
Done